src/main/java/core/basesyntax/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.report;
+
+import core.basesyntax.service.ShopService;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private ShopService shopService;

```

Comment:
It's better to provide `ShopService` via constructor to ensure that `ReportGeneratorImpl` always has a non-null instance of `ShopService`. This way, you can avoid null checks and NullPointerExceptions. This is a part of Dependency Injection principle.